lib/Conversion/ATenToXTenPass.cpp

```diff
@@ -47,33 +47,67 @@
 #define DEBUG_TYPE "aten-to-xten-pass"
 
 using namespace mlir;
-using namespace xilinx::xten;
+// using namespace xilinx::xten;
+using namespace xilinx;
+using namespace mlir::torch;
 
 namespace {
 
+long getInputSize(xten::Conv2dOp &c2d) {
+  TensorType inputType = c2d.input().getType().dyn_cast<Torch::ValueTensorType>().toBuiltinTensor();
+  if (inputType == nullptr || !inputType.hasRank())
+    return 0;
+  auto shape = inputType.getShape();
+  // multiply all dimensions together
+  long result = 1;
+  for (auto it = shape.begin(); it != shape.end(); it++)
+    result *= *it;
+  return result;
+}
+
+// pick conv2d with smallest input size.
+bool fuseFirstC2dInTensorAdd(xten::Conv2dOp &c2d0, xten::Conv2dOp &c2d1) {
+  auto s0 = getInputSize(c2d0);
+  auto s1 = getInputSize(c2d1);  
+  return s0 && s1 && s0 < s1;
+}
+// adapter fun because tablegen can only bind result and not the op itself.
+bool fuseFirstC2dInTensorAdd(OpResult a, OpResult b) {
+  auto c2d0 = cast<xten::Conv2dOp>(a.getOwner());
+  auto c2d1 = cast<xten::Conv2dOp>(b.getOwner());
+  return fuseFirstC2dInTensorAdd(c2d0, c2d1);
+}
+
+namespace atenToXten {
 #include "xten/Conversion/ATenToXTen.cpp.inc"
+}
 
-struct ATenToXTenPass : public ATenToXTenBase<ATenToXTenPass> {
+namespace xtenToXtenCleanup {
+#include "xten/Conversion/ATenToXTen2ndRound.cpp.inc"
```

Comment:
Why is the second round needed? Can we move this into `ATenToXTen` or rename to something more appropriate like `XTenToXtenPass`?

Reply:
I tried to explain this in the PR description, though I should move that into comments. I think XTenToXTenPass is not a very good name as it is not a pass per se, and the name doesn't describe what the rewriting does... maybe XTenFusions or so?

Reply:
`XTenFusions` and some description would be fine with me.